plugins/steve/resource-instance.js

```diff
@@ -1250,19 +1250,18 @@ export default {
           if (isString(pathValue)) {
             pathValue = pathValue.trim();
           }
-
           if (requiredIfPath) {
             const reqIfVal = get(data, requiredIfPath);
 
-            if (!isEmpty(reqIfVal) && isEmpty(pathValue)) {
+            if (!isEmpty(reqIfVal) && (isEmpty(pathValue) && pathValue !== 0)) {
               errors.push(this.t('validation.required', { key: displayKey }));
             }
           }
 
           validateLength(pathValue, rule, displayKey, this.$rootGetters, errors);
           validateChars(pathValue, rule, displayKey, this.$rootGetters, errors);
 
-          if ( !isEmpty(pathValue) && DNS_LIKE_TYPES.includes(fieldType) ) {
+          if ( isEmpty(pathValue) && DNS_LIKE_TYPES.includes(fieldType) ) {
```

Comment:
Seems like the `!` was removed here accidentally?  If it's empty then lowercasing it isn't going to do anything.

Reply:
Yep you're right. Re-added it